Signal/src/AppDelegate.m

```diff
@@ -592,6 +593,9 @@ - (void)applicationDidBecomeActive:(UIApplication *)application {
         [self handleActivation];
     }];
 
+    // We want to process up to one local notification per activation, so clear the flag.
```

Comment:
Can you clarify for me - what problem are you trying to solve here (from the user's perspective) with this code?

Reply:
The symptom is: you launch Signal from a local notification for an incoming message.  

Expected: Signal "pops/closes" any presented views and pushes the conversation view for that message's thread - unless that thread is already "frontmost".  
Observed: Signal will do that "pop and present" logic N times - you enter convo view, pop to home view, enter convo view, return to home view, etc.  This issue is very hard to repro although I've seen it myself a couple times.  I decided to add a bunch of logging so that we could debug this without a repro and it paid off.  Riya repro'd and the logging clearly shows the app receiving multiple (identical) notifications (via `didReceiveLocalNotification`) during the activation process (after foreground, before active).  

Also: N can be quite high, like 5-10.  N seems vaguely related to their being multiple signal notifications, but the repro isn't that simple.